server/audit/views.py

```diff
@@ -1,69 +1,202 @@
         serializer_class = AuditSerializer
-        if self.action in ['retrieve']:
+        if self.action in ['list', 'retrieve']:
             serializer_class = GetAuditSerializer
         return serializer_class(*args, **kwargs)
 
     def list(self, request):
-        if request.GET.get("status", '') != '':
-            queryset = Audit.objects.filter(status=request.GET.get("status", '')) \
-                .filter(organization_id=request.GET.get("organization", ''))
-        else:
-            queryset = Audit.objects.filter(organization_id=request.GET.get("organization", ''))
+        org_id = request.query_params.get('organization')
+        audit_status = request.query_params.get('status')
+        assigned_sk = request.query_params.get('assigned_sk')
+
+        if audit_status:
+            self.queryset = self.queryset.filter(status=audit_status)
+        if org_id:
+            self.queryset = self.queryset.filter(organization_id=org_id)
+        if assigned_sk:
+            self.queryset = self.queryset.filter(assigned_sk__id=assigned_sk)
+
+        serializer = self.get_serializer(self.queryset, many=True)
+        return Response(serializer.data)
+
+    def get_item_serializer(self, *args, **kwargs): # pylint: disable=no-self-use 
+        serializer_class = ItemSerializer
+        return serializer_class(*args, **kwargs)
+
+    # pylint: disable=protected-access
+    @action(detail=False, methods=['GET'], name='Check Item for Validation')
+    def check_item(self, request):
+        bin_id = request.query_params.get('bin_id')
+        audit_id = request.query_params.get('audit_id')
+        item_id = request.query_params.get('item_id')
+
+        try: # Check that the item exists for this Audit
+            bins = BinToSK.objects.get(bin_id=bin_id)
+            audit = Audit.objects.get(audit_id=audit_id)
+            item = audit.inventory_items.get(_id=item_id)
+        except (ObjectDoesNotExist, ValueError) as ex:
+            raise Http404 from ex
+
+        record = {}
+        try: # Check that the item hasn't already been recorded
+            record = Record.objects.get(item_id=item_id)
+        except ObjectDoesNotExist:
+            if item._id in bins.item_ids: # Check that the item belongs to the current bin
+                serializer = self.get_item_serializer(item, many=False)
+                response = Response(serializer.data)
+            else:
+                response = Response({
+                'detail': 'Item part of Audit but not of Bin',
+                'inAudit': audit_id}, status=status.HTTP_400_BAD_REQUEST)
 
-        serializer = self.get_serializer(queryset, many=True)
+        if record and str(record.audit.audit_id) == str(audit_id):
+            response = Response({
+            'detail': 'This item has already been completed',
+            'alreadyMatched': record.record_id},
+            status=status.HTTP_400_BAD_REQUEST)
+
+        return response
```

Comment:
we should add a default response just to be safe, this was done to avoid multiple return statements

Reply:
We can fix this with a default case

Reply:
yeah exactly, I just found a bug where if you have 2 audits that contain the same item and a stock keeper tries to complete the item, it will raise a server error as a response. We should be able to handle that case.